ts/services/PackageService.ts

```diff
@@ -102,6 +103,50 @@ class PackageService {
       logger.error('There was an error uninstalling the package: ' + e);
     }
   }
+
+  public async createArchive(location: string, config, logger): Promise<void> {
+    logger.debug('\nCreating archive package ');
+    const output = fs.createWriteStream(location + '/pkg.zip');
+    const archive = archiver('zip', {
+      zlib: { level: 9 } // Sets the compression level.
+    });
+
+    // listen for all archive data to be written
+    // 'close' event is fired only when a file descriptor is involved
+    output.on('close', () =>  {
+      logger.info(archive.pointer() + ' total bytes');
+      logger.info('archiver has been finalized and the output file descriptor has closed.');
+    });
+
+    // This event is fired when the data source is drained no matter what was the data source.
+    // It is not part of this library but rather from the NodeJS Stream API.
+    // @see: https://nodejs.org/api/stream.html#stream_event_end
+    output.on('end', () => {
+      logger.info('Data has been drained');
+    });
+
+    // good practice to catch warnings (ie stat failures and other non-blocking errors)
+    archive.on('warning', (err) =>  {
+      if (err.code === 'ENOENT') {
+        logger.warn(err);
+      } else {
+        // throw error
+        throw err;
+      }
+    });
+
+    // good practice to catch this error explicitly
+    archive.on('error', (err) =>  {
+      throw err;
+    });
+
+    // pipe archive data to the file
+    archive.pipe(output);
+
+    archive.directory('./', 'pkg-dir');
+    archive.finalize();
+
```

Comment:
Trying to find a TSLint rule to catch this trailing new line in a function, but can't find one.

Reply:
I removed it